backend/coins/eth/etherscan/etherscan.go

```diff
@@ -67,9 +68,13 @@ func (etherScan *EtherScan) call(params url.Values, result interface{}) error {
 	if response.StatusCode != http.StatusOK {
 		return errp.Newf("expected 200 OK, got %d", response.StatusCode)
 	}
-	if err := json.NewDecoder(response.Body).Decode(result); err != nil {
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
 		return errp.WithStack(err)
 	}
+	if err := json.Unmarshal(body, result); err != nil {
+		return errp.Newf("unexpected response: %s", string(body))
```

Comment:
Since this is visible to the user, I would clarify:

> unexpected response from Etherscan: %s

Reply:
done